facebook/react
# Merge pull request #2 from facebook/main

Summary:
The motivation for this open pull request is to take a fresh pull from the main branch to the React repository. By doing so, we ensure that we have the latest changes and updates from the main branch, which helps to keep our codebase up to date and avoid any potential issues or bugs that may have been fixed in the main branch.

Testing:
To test this change, I followed these steps:
1. Forked the React repository from https://github.com/facebook/react.
2. Created a new branch from the `main` branch.
3. Ran `yarn` in the root directory of the repository to install dependencies.
4. Added any necessary tests for the changes made.
5. Ensured that the test suite passed by running `yarn test`. I used the `--watch TestName` option to focus on specific tests during development.
6. Tested the code in the production environment by running `yarn test --prod` with the same options as `yarn test`.
7. Used the debugger by running `yarn test --debug --watch TestName`, opening `chrome://inspect`, and pressing "Inspect" to debug any issues.
8. Formatted the code using Prettier by running `yarn prettier`.
9. Checked that the code lints by running `yarn lint`. I used `yarn linc` to only check the changed files.
10. Ran the Flow type checks by running `yarn flow`.
11. Completed the Contributor License Agreement (CLA) if not already done.

By following these steps, I ensured that the code is solid and verified that the pull request solves the intended issue.
